src/main/cpp/t07_ip.cpp

```diff
@@ -34,5 +34,37 @@
 using namespace std;
 
 int t07_ip() {
+    char c;
+    int numOfDots = 0;
+    string num ="";
 
+    while(cin.get(c)){
+        //cout << c << " " << int(c) << endl;
+        if(int(c) < 48 || int(c) > 57 ){
+            if(int(c) != 46){
+                cout << "NO";
+                return 0;
+            }
+        }
+        if(int(c) == 46){
+            if(num.length() == 0|| stoi(num) > 255 || numOfDots == 3 || (num.length() > 1 && char(num[0]) == '0')){
+                cout << "NO";
+                return 0;
+            }
+            numOfDots ++;
+            num = "";
+        }else{
+            num += c;
+        }
+        //cout << num << endl;
+    }
+
+    if(num.length() == 0 || stoi(num) > 255 || numOfDots != 3){
+        cout << "NO";
+        return 0;
+    }else{
+        numOfDots ++;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'numOfDots' is assigned a value that is never used.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4678874)